src/bech32.cpp

```diff
@@ -147,11 +148,15 @@ namespace bech32
 std::string Encode(const std::string& hrp, const data& values) {
     data checksum = CreateChecksum(hrp, values);
     data combined = Cat(values, checksum);
-    std::string ret = hrp + '1';
-    ret.reserve(ret.size() + combined.size());
+    size_t hrp_size = hrp.size();
+    char ret[hrp_size + combined.size() + 2];
+    memcpy(ret, hrp.c_str(), hrp_size);
```

Comment:
Is there slow-down associated with inlining `hrp_size`? Seems the compiler would optimize that.

Reply:
I'll check. I didn't like repeatedly calling hrp.size() but it may be optimized away, as you say.

Reply:
I don't think the difference warrants the extra line, so I removed it. Thanks for the nudge. :)